internal/data/inmemory/carts/carts.go

```diff
@@ -12,44 +11,57 @@ import (
 )
 
 type Carts struct {
-	idCart atomic.Int64
-	rw     sync.RWMutex
-	m      map[int64]cart.Cart
+	idCart    atomic.Int64
+	rw        sync.RWMutex
+	idCartMap map[int64]*cart.Cart
+	ownerMap  map[int64]*cart.Cart
 }
 
-// Точно не пустой!
 func New() *Carts {
 	return &Carts{
-		m: make(map[int64]cart.Cart),
+		idCartMap: make(map[int64]*cart.Cart),
+		ownerMap:  make(map[int64]*cart.Cart),
 	}
 }
 
-func (c *Carts) Get(ctx context.Context, idOwner int64) (cart.Cart, error) {
+func (c *Carts) GetByCartId(ctx context.Context, cartId int64) (cart.Cart, error) {
 	c.rw.RLock()
 	defer c.rw.RUnlock()
```

Comment:
Логическое отделение блоков 


```suggestion
	c.rw.RLock()
	defer c.rw.RUnlock()
	
	cartNew, ok := c.idCartMap[cartId]
```

Reply:
А что тут конкретно не отделено? Мьютексы идут рядом друг с другом. Добавление в мапы отдельно.

Reply:
Ты ниже сказал, что все ок с методом)

Reply:
Ц, я не увидел Suggested change. Думал что ну вот же оно отделено) Поправил в коде